openedx/core/djangoapps/public_api/courses/tests/test_course_import_export.py

```diff
@@ -0,0 +1,416 @@
+from django.conf import settings
+from django.core.urlresolvers import reverse
+
+from xmodule.modulestore.tests.factories import ItemFactory, CourseFactory
+
+from .utils import CourseTestCase
+from student import auth
+from student.roles import CourseInstructorRole, CourseStaffRole
+
+TEST_DATA_CONTENTSTORE = copy.deepcopy(settings.CONTENTSTORE)
+TEST_DATA_CONTENTSTORE['DOC_STORE_CONFIG']['db'] = 'test_xcontent_{}'.format(
+    uuid4().hex
+)
+
+log = logging.getLogger(__name__)
+
+
+def course_url(handler, course_key, kwargs=None):
+    """
+    Reverse a handler that uses a course key.
+
+    :param handler: a URL handler name
+    :param course_key: a CourseKey
+    :return: the reversed URL string of the handler with the given course key
+    """
+    kwargs_for_reverse = {'course_key_string': course_key.id}
+    if kwargs:
+        kwargs_for_reverse.update(kwargs)
+
+    return reverse(
+        handler,
+        kwargs=kwargs_for_reverse
+    )
+
+
+class CourseListTestCase(CourseTestCase):
+    """
+    Unit tests for course listing
+    """
+    def setUp(self):
+        super(CourseListTestCase, self).setUp()
+        self.url = reverse('course_listing_handler')
+
+    def test_one_course(self):
+        """
+        Test that the course list view correctly returns courses.
+        """
+        result = self.client.get(self.url, HTTP_ACCEPT='application/json')
+
+        self.assertListEqual(
+            json.loads(result.content)['courses'],
+            [unicode(self.course.location.course_key)]
+        )
+
+    def test_multiple_courses(self):
+        """
+        Test that multiple courses are properly listed by the view.
+        """
+        CourseFactory.create(
+            org="MITx",
+            course="FabulousCourse",
+            run="Spring_2015"
+        )
+        result = self.client.get(self.url, HTTP_ACCEPT='application/json')
+        result_courses = json.loads(result.content)['courses']
+
+        self.assertIn(unicode(self.course.location.course_key), result_courses)
+        self.assertIn(u'MITx/FabulousCourse/Spring_2015', result_courses)
+
+    def test_session_authentication(self):
+        """
+        Verify that session authenticated users can make requests.
+        """
+        client = Client()
+        client.login(username=self.user.username, password=self.user_password)
+        result = client.get(self.url, HTTP_ACCEPT='application/json')
+
+        self.assertListEqual(
+            json.loads(result.content)['courses'],
+            [unicode(self.course.location.course_key)]
+        )
+
+
+@override_settings(CONTENTSTORE=TEST_DATA_CONTENTSTORE)
+class ImportTestCase(CourseTestCase):
+    """
+    Unit tests for importing a course
+    """
+    def setUp(self):
+        super(ImportTestCase, self).setUp()
+        self.url = course_url('course_import_export_handler', self.course)
+        self.content_dir = path(tempfile.mkdtemp())
+
+        def touch(name):
```

Comment:
If you use [`path.py`](https://pythonhosted.org/path.py/api.html), this and other useful operations are built-in as methods. It would simplify this test a lot.